### DISCUSSION TITLE
Simple Scripts Discussion

### DISCUSSION THREAD

**Discussion by TheGassyNinja** [BreadOnPenguins/scripts#1] [Ideas]
I would like to share this script that I have been using for a few years to take screenshots WITHOUT the overhead of extra utilities. It uses Imagemagic and offers 2 options. A full screenshot || selected area shot.

```shell
#!/usr/bin/env bash
export DISPLAY=:0
current=$(date +%H-%M-%S-%d-%m-%Y).png
if [[ -z "${1}" ]]; then
   import -window root "${HOME}/${current}" # All screen
else
   import "${HOME}/${current}" # Custom area screenshot
fi
notify-send "Screenshot ${current} taken successfully!"
``` 

Any constructive criticism is welcome. 
What simple scripts do you have that fit with Breads script collection?

**Comment by BreadOnPenguins** [comment]
Thanks so much for sharing, and sorry for the late reply!
Tried it myself and it works perfectly. 
Would it be alright if I cover this in a video at some point, and also go ahead and add it into the repo? Just wanted to double check before.

**Reply by TheGassyNinja** [reply]
Of course you can cover it and add it in.

**Reply by BreadOnPenguins** [reply]
Much appreciated :)

**Comment by panv-00** [comment]
great script.. and very useful... thank you for sharing.
I had one issue though.. due to some settings in my system, the mouse was leaving a trail (check below):


![13-58-32-18-06-2025](https://github.com/user-attachments/assets/10f6a290-664e-4012-ae90-12ad9ba3989b)

so, I added a `sleep 1` between `current ` and `if `:

```bash
#!/usr/bin/bash

export DISPLAY=:0
current=$(date +%H-%M-%S-%d-%m-%Y).png
sleep 1 # adding this sleep made the mouse trail go away
if [[ -z "${1}" ]]; then
  import -window root "${HOME}/${current}" # All screen
else
  import "${HOME}/${current}" # Custom area screenshot
fi
notify-send "Screenshot ${current} taken successfully!"
```

that made the trail go away....


![13-58-54-18-06-2025](https://github.com/user-attachments/assets/7c16654b-48ef-4d17-a363-739a3b232720)

**Reply by gophermail** [reply]
op make any tweaks to this since 2024??

i just changed superficial stuff like date format and directories

i also didnt have imagemagic so i had to install it, and made a comment so i dont forget if i use it on a new computer

i also set up keybindings for prtsc to print whole window and windows+prtsc to do select area

    gopher@gopherbox1:~/.local/bin/screenshot$ cat screenshot.sh
    #!/usr/bin/env bash
    
    #depends on imagemagick
    export DISPLAY=:0
    current=$(date +%Y-%m-%d-%H:%M:%S).png
    if [[ -z "${1}" ]]; then
        import -window root "/home/gopher/screenshots/${current}" # All screen
    else
       import "/home/gopher/screenshots/${current}" # Custom area screenshot
    fi
    notify-send "Please be advised, CIA capture ${current} uploaded successfully!"

**Reply by panv-00** [reply]
I added:
```xclip -selection clipboard -t image/png -i "$screenshot_path"```
before notify-send to also put the screenshot in my clipboard, that way i can paste it anywhere if i needed to

**Reply by gophermail** [reply]
oh smart
